invokeai/app/invocations/prompt_tools.py

```diff
@@ -0,0 +1,286 @@
+
+
+@invocation(
+    "prompt_split_neg",
+    title="Prompt Split Negative",
+    tags=["prompt", "split", "negative"],
+    category="prompt",
+    version="1.0.0",
+)
+class PromptSplitNegInvocation(BaseInvocation):
+    """Splits prompt into two prompts, inside [] goes into negative prompt everthing else goes into positive prompt. Each [ and ] character is replaced with a space"""
+
+    prompt: str = InputField(default="", description="Prompt to split")
+
+    def invoke(self, context: InvocationContext) -> PromptPosNegOutput:
+        p_prompt = ""
+        n_prompt = ""
+        brackets_depth = 0
+        escaped = False
+
+        for char in self.prompt or "":
+            if char == "[" and not escaped:
+                n_prompt += " "
+                brackets_depth += 1
+            elif char == "]" and not escaped:
+                brackets_depth -= 1
+                char = " "
+            elif brackets_depth > 0:
+                n_prompt += char
+            else:
+                p_prompt += char
+
+            # keep track of the escape char but only if it isn't escaped already
+            if char == "\\" and not escaped:
+                escaped = True
+            else:
+                escaped = False
+
+        return PromptPosNegOutput(positive_prompt=p_prompt, negative_prompt=n_prompt)
+
+
+@invocation("prompt_join", title="Prompt Join", tags=["prompt", "join"], category="prompt", version="1.0.0")
+class PromptJoinInvocation(BaseInvocation):
+    """Joins prompt left to prompt right"""
+
+    prompt_left: str = InputField(default="", description="Prompt Left", ui_component=UIComponent.Textarea)
+    prompt_right: str = InputField(default="", description="Prompt Right", ui_component=UIComponent.Textarea)
+
+    def invoke(self, context: InvocationContext) -> StringOutput:
+        return StringOutput(value=((self.prompt_left or "") + (self.prompt_right or "")))
+
+
+@invocation("prompt_join_three", title="Prompt Join Three", tags=["prompt", "join"], category="prompt", version="1.0.0")
+class PromptJoinThreeInvocation(BaseInvocation):
+    """Joins prompt left to prompt middle to prompt right"""
+
+    prompt_left: str = InputField(default="", description="Prompt Left", ui_component=UIComponent.Textarea)
+    prompt_middle: str = InputField(default="", description="Prompt Middle", ui_component=UIComponent.Textarea)
+    prompt_right: str = InputField(default="", description="Prompt Right", ui_component=UIComponent.Textarea)
+
+    def invoke(self, context: InvocationContext) -> StringOutput:
+        return StringOutput(value=((self.prompt_left or "") + (self.prompt_middle or "") + (self.prompt_right or "")))
+
+
+@invocation(
+    "prompt_replace", title="Prompt Replace", tags=["prompt", "replace", "regex"], category="prompt", version="1.0.0"
+)
+class PromptReplaceInvocation(BaseInvocation):
+    """Replaces the search string with the replace string in the prompt"""
```

Comment:
Should just be a generic StringRegexReplaceInvocation.

Reply:
agree if the others are moving the name from prompt to string then this should be the same and also move into strings.py

Reply:
Name changed to StringReplace and moved into strings.py . Didn't put regex in the name as that is an option that is off as default and might confuse some users.